x-pack/plugins/security_solution/public/detections/components/rules/eql_query_bar/validators.ts

```diff
@@ -67,8 +67,17 @@ export const eqlValidator = async (
 
   try {
     const { data } = KibanaServices.get();
+    let dataViewTitle = index?.join();
+    let runtimeMappings = {};
+    if (dataViewId != null) {
+      const dataView = await data.dataViews.get(dataViewId);
```

Comment:
Discussed outside of this thread but yes wholeheartedly agree with storing the data view object to redux. I think this can fall into the same category that updating where / how the `useFetchIndex` hook is used.